tests/cephfs/cephfs_nfs/test_cephfs_nfs_ha_keeplive_services.py

```diff
@@ -0,0 +1,184 @@
+
+            log.info("ceph nfs export created successfully")
+            out, rc = client1.exec_command(
+                sudo=True, cmd=f"ceph nfs export get {nfs_name} {nfs_export}"
+            )
+            json.loads(out)
+            mounting_dir = "".join(
+                random.choice(string.ascii_lowercase + string.digits)
+                for _ in list(range(10))
+            )
+            nfs_mounting_dir = f"/mnt/cephfs_nfs{mounting_dir}_1/"
+            client1.exec_command(sudo=True, cmd=f"mkdir -p {nfs_mounting_dir}")
+            command = f"mount -t nfs -o port=2049 {virtual_ip}:{nfs_export} {nfs_mounting_dir}"
+            client1.exec_command(sudo=True, cmd=command, check_ec=False)
+            mount_dir.append(nfs_mounting_dir)
+
+        backend_server = fs_util_v1.get_active_nfs_server(
+            client1, nfs_name, ceph_cluster
+        )
+        haproxy_ls = fs_util_v1.get_daemon_status(client1, "haproxy")
+        keepalived_ls = fs_util_v1.get_daemon_status(client1, "keepalived")
+        haproxy_ls_hostnames = [entry["hostname"] for entry in haproxy_ls]
+        keepalived_ls_hostnames = [entry["hostname"] for entry in keepalived_ls]
+        log.info(backend_server)
+        for server in backend_server:
+            log.info(f"Active backend servers {server.node.hostname}")
+        dir_name = "smallfile_dir"
+        for i, mount in enumerate(mount_dir):
+            client1.exec_command(sudo=True, cmd=f"mkdir -p {mount}{dir_name}_{i}")
+        with parallel() as p:
+            p.spawn(fs_util_v1.write_io, client1, mount_dir)
+            for i in backend_server:
+                fs_util_v1.reboot_node(i, timeout=300)
+                time.sleep(120)
+                fs_util_v1.wait_for_host_online(client1, i)
+
+        backend_server = fs_util_v1.get_active_nfs_server(
+            client1, nfs_name, ceph_cluster
+        )
+        haproxy_ls_after = fs_util_v1.get_daemon_status(client1, "haproxy")
+        keepalived_ls_after = fs_util_v1.get_daemon_status(client1, "keepalived")
+        haproxy_ls_after_hostnames = [entry["hostname"] for entry in haproxy_ls_after]
+        keepalived_ls_after_hostnames = [
+            entry["hostname"] for entry in keepalived_ls_after
+        ]
+        if (
+            haproxy_ls_after_hostnames == haproxy_ls_hostnames
+            or keepalived_ls_after_hostnames == keepalived_ls_hostnames
+        ):
+            raise CommandFailed(
+                f"The ha proxy node did not change even after power off ."
+                f"\nPrevious haproxy nodes: {haproxy_ls_hostnames}"
+                f"\nCurrent haproxy nodes: {haproxy_ls_after_hostnames}"
+                f"\nPrevious keepalived nodes: {keepalived_ls_hostnames}"
+                f"\nCurrent keepalived nodes: {keepalived_ls_after_hostnames}"
+            )
+        log.info(
+            f"The ha proxy node did not change even after power off ."
```

Comment:
Sentence needs a change? both failure and pass case have same statements

Reply:
Updated